tools/run_tests/xds_k8s_test_driver/framework/test_app/runners/k8s/k8s_xds_server_runner.py

```diff
@@ -193,23 +190,44 @@ def run(  # pylint: disable=arguments-differ,too-many-branches
         # Verify the deployment reports all pods started as well.
         self._wait_deployment_with_available_replicas(self.deployment_name,
                                                       replica_count)
-        servers = []
+        servers: List[XdsTestServer] = []
         for pod in pods:
-            if self.debug_use_port_forwarding:
-                pf = self._start_port_forwarding_pod(pod, maintenance_port)
-                rpc_port, rpc_host = pf.local_port, pf.local_address
-            else:
-                rpc_port, rpc_host = maintenance_port, None
-
             servers.append(
-                XdsTestServer(ip=pod.status.pod_ip,
-                              rpc_port=test_port,
-                              hostname=pod.metadata.name,
-                              maintenance_port=rpc_port,
-                              secure_mode=secure_mode,
-                              rpc_host=rpc_host))
+                self._xds_test_server_for_pod(pod,
+                                              test_port=test_port,
+                                              maintenance_port=maintenance_port,
+                                              secure_mode=secure_mode))
         return servers
 
+    def _get_default_maintenance_port(self, secure_mode: bool) -> int:
+        if not secure_mode:
+            maintenance_port = self.DEFAULT_MAINTENANCE_PORT
+        else:
+            maintenance_port = self.DEFAULT_SECURE_MODE_MAINTENANCE_PORT
+        return maintenance_port
+
+    def _xds_test_server_for_pod(self,
+                                 pod: k8s.V1Pod,
+                                 *,
+                                 test_port: int = DEFAULT_TEST_PORT,
+                                 maintenance_port: Optional[int] = None,
+                                 secure_mode: bool = False) -> XdsTestServer:
+        if maintenance_port is None:
```

Comment:
[Nit] `if not maintenance_port:`

Reply:
Here it was on purpose to allow port value `0`. This is normally a legit value, meaning the port should be randomized. However, since we don't support randomization of this port, I agree with your suggestion.